synapseclient/models/fileview.py

```diff
@@ -0,0 +1,1104 @@
+import dataclasses
+from collections import OrderedDict
+from copy import deepcopy
+from dataclasses import dataclass, field
+from datetime import date, datetime
+from typing import Any, Dict, List, Optional, Protocol, Set, TypeVar, Union
+
+from typing_extensions import Self
+
+from synapseclient import Synapse
+from synapseclient.api import ViewTypeMask
+from synapseclient.core.async_utils import async_to_sync
+from synapseclient.core.constants import concrete_types
+from synapseclient.core.utils import MB, delete_none_keys
+from synapseclient.models import Activity, Annotations
+from synapseclient.models.mixins.access_control import AccessControllable
+from synapseclient.models.mixins.table_components import (
+    ColumnMixin,
+    DeleteMixin,
+    GetMixin,
+    QueryMixin,
+    ViewBase,
+    ViewSnapshotMixin,
+    ViewStoreMixin,
+    ViewUpdateMixin,
+)
+from synapseclient.models.table_components import Column
+
+DATA_FRAME_TYPE = TypeVar("pd.DataFrame")
+
+
+class FileViewSynchronousProtocol(Protocol):
```

Comment:
I wasn't planning on Dataset having a sync protocol as we discussed in Slack, should I switch to follow this pattern?